src/Interpreters/Context.cpp

```diff
@@ -1035,7 +1035,21 @@ void Context::setTemporaryStoragePolicy(const String & policy_name, size_t max_s
 
 void Context::setTemporaryStorageInCache(const String & cache_disk_name, size_t max_size)
 {
-    auto disk_ptr = getDisk(cache_disk_name);
+    String config_prefix = "storage_configuration.disks";
+    const auto & config = getConfigRef();
+    auto disk_selector = std::make_shared<DiskSelector>();
+    /** We do not want to initialize all disks, so we filter only those that are used for temporary storage.
+      * This includes the cache disk itself and the underlying disk.
+      * In some cases, this may not work if we have more than one level of nesting, but disks3 + cache should be enough.
```

Comment:
I do not see why should we do this, because afaik server must fail if any disk cannot initialize, because no disk can be accessed unless all disks are initialized. For example, I've seen many times that when my aws token expires, the server always fails to start with expired token exception from initialization of disk s3.